### PR TITLE
Transactions

### PR DESCRIPTION
**Pull request by piotrkowalczuk**
This PR is a proposal of how transactions support can be implemented.

### CODE REVIEW

File: example/app/internal/model/schema_news_test.go [hunk 1]
```diff
@@ -756,3 +755,39 @@ func TestNewsRepositoryBase_UpsertQuery(t *testing.T) {
 		})
 	}
 }
+
+func TestNewsRepositoryBase_InsertTx(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (karthikraobr) [reply example/app/internal/model/schema_news_test.go]: Looks good to me.

File: example/app/internal/model/schema_news_test.go [hunk 2]
```diff
@@ -756,3 +756,117 @@ func TestNewsRepositoryBase_UpsertQuery(t *testing.T) {
...
+
+	for _, id := range ids {
+		if _, err := s.news.FindOneByID(ctx, id); err != nil {
+			t.Fatalf("unexpected error: %s", err.Error())
+		} else {
+			t.Logf("news with id %d found", id)
+		}
+	}
+}
+func TestNewsRepositoryBase_RunInTransaction(t *testing.T) {
```
Comment (piotrkowalczuk) [inline example/app/internal/model/schema_news_test.go]: @ders @karthikraobr Here you can find `RunInTransaction` example usage. Let me know what do you think.

File: example/app/internal/model/schema_news_test.go [hunk 3]
```diff
@@ -756,3 +756,117 @@ func TestNewsRepositoryBase_UpsertQuery(t *testing.T) {
...
 	}
 }
+
+func TestNewsRepositoryBaseTx_Insert(t *testing.T) {
+	s := setup(t)
+	defer s.teardown(t)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 50*time.Second)
+	defer cancel()
+	tx, err := s.news.BeginTx(ctx)
```
Comment (piotrkowalczuk) [inline example/app/internal/model/schema_news_test.go]: @ders Separate API was replaced by additional `struct` that has similar API but each statement is executed within the scope of a transaction.

File: example/app/internal/model/schema_news_test.go (hunk 2 above)
Comment (karthikraobr) [reply example/app/internal/model/schema_news_test.go]: Looks good to me

**Pull request merged**